src_py/sprite.py

```diff
@@ -1695,25 +1698,39 @@ def spritecollide(sprite, group, dokill, collided=None):
         for group_sprite in group.sprites():
             if collided is not None:
                 if collided(sprite, group_sprite):
-                    group_sprite.kill()
-                    append(group_sprite)
+                    if not ignore_self or sprite != group_sprite:
+                        group_sprite.kill()
+                        append(group_sprite)
             else:
                 if default_sprite_collide_func(group_sprite.rect):
-                    group_sprite.kill()
-                    append(group_sprite)
+                    if not ignore_self or sprite != group_sprite:
+                        group_sprite.kill()
+                        append(group_sprite)
 
         return crashed
 
     if collided is not None:
+        if ignore_self:
+            return [
+                group_sprite for group_sprite in group if collided(sprite, group_sprite) and sprite != group_sprite
```

Comment:
```suggestion
                group_sprite for sprite != group_sprite and group_sprite in group if collided(sprite, group_sprite) 
```

I think it will be ever so slightly faster to do the self check first as short circuit evaluation will cause it to skip the collision check if the left hand side of the and fails and the collision check is generally going to be slower. See:

https://en.wikipedia.org/wiki/Short-circuit_evaluation